src/App.tsx

```diff
@@ -1,35 +1,35 @@
 import React from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
+import { Search } from './Search';
 import moviesFromServer from './api/movies.json';
 
-type State = {};
+type State = {
+  query: string;
+};
 
 export class App extends React.Component<{}, State> {
-  state: State = {};
+  state: State = {
+    query: '',
+  };
+
+  onChangeFunction = (value: string): void => {
+    this.setState({ query: value });
+  };
 
   render() {
+    const { query } = this.state;
+    const movies = [...moviesFromServer].filter(({ title, description }) => (
```

Comment:
what does filter return? is it mutating method?
```suggestion
    const movies = moviesFromServer.filter(({ title, description }) => (
```

Reply:
you are right